### PR TITLE
Add SSM parameter store functionality to dss-ops

### LINKED ISSUE #2299: Create operational tooling for Lambda SSM environment variables
The functionality of `scripts/populate_lambda_ssm_parameters.py` should be moved into the DSS operational framework, providing capability to:
  - [x] Print the current parameters in the SSM store
  - [x] Push new parameters into the SSM store
  - [ ] Print the current environment variables in the deployed Lambda
  - [ ] Set a single variable into each deployed Lambda
  - [ ] Remove a single variable from each deployed Lambda

### PR DESCRIPTION
**Pull request by chmreid**
This PR adds functionality to the `dss-ops.py` script that allows operators to list, set, and unset parameters stored in the SSM store under the "environment" variables. It also adds tests for the new SSM parameter functionality.

This PR is part of #2299 and is a reworked version of (part of) a PR that was abandoned (#2324). This PR is much simpler and should be easier/faster to review.

### CODE REVIEW

File: dss/operations/ssm_params.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+import typing
+
+from botocore.exceptions import ClientError
+
+from dss.operations import dispatch
+from dss.util.aws.clients import secretsmanager  # type: ignore
+import dss.operations.util as util
+
+from dss.util.aws.clients import ssm as ssm_client  # type: ignore
+from dss.util.aws.clients import secretsmanager as sm_client  # type: ignore
```
Comment (chmreid) [inline dss/operations/ssm_params.py]: FIXME: just noticed secretsmanager is imported twice

File: dss/operations/ssm_params.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+def set_ssm_parameter(env_var: str, value) -> None:
+    """
+    Set a parameter in the SSM param store variable "environment".
+
+    :param env_var: the name of the environment variable being set
+    :param value: the value of the environment variable being set
+    """
+    environment = get_ssm_environment()
+    try:
```
Comment (natanlao) [inline dss/operations/ssm_params.py]: Is there a reason this is not `prev_value = environment.get(env_var)`?

File: tests/__init__.py [hunk 3]
```diff
@@ -160,3 +160,31 @@ def __exit__(self, *args, **kwargs):
         # Clean up by setting sys.stdout back to what it was before we opened
         # up this context
         sys.stdout = self._stdout
+
+class SwapStdin(object):
+    """Utility object using a context manager to swap out stdin with user-provided data."""
+    def __init__(self, *, input=None):
```
Comment (natanlao) [inline tests/__init__.py]: `input` is a builtin, you may want to consider renaming the argument

File: tests/__init__.py [hunk 4]
```diff
@@ -160,3 +160,31 @@ def __exit__(self, *args, **kwargs):
...
+        To swap out stdin properly (so it still works with the select module) requires a "real" file
+        with an os-level file descriptor. Fortunately os.pipe() will create a file descriptor for
+        the pipe, so we create a pipe and fill it with mock data, then swap it out with stdin.
+        """
+        fdr, fdw = os.pipe()
+        os.write(fdw, self.input)
+        os.close(fdw)
+        f = os.fdopen(fdr)  # use the file descriptor directly
+
+        self._stdin = sys.stdin
```
Comment (natanlao) [inline tests/__init__.py]: Nit: the original `stdin` (at start of program execution) is also stored in `sys.__stdin__`

File: tests/test_operations.py [hunk 5]
```diff
@@ -423,6 +424,78 @@ def paginate(self):
...
+        testvar_value2 = "Goodbye world!"
+
+        # Assemble an old and new environment to return
+        old_env = {"dummy_key": "dummy_value"}
+        new_env = dict(**old_env)
+        new_env[testvar_name] = testvar_value
+        ssm_old_env = _wrap_ssm_env(old_env)
+        ssm_new_env = _wrap_ssm_env(new_env)
+
+        with self.subTest("Create a new SSM parameter"):
```
Comment (natanlao) [inline tests/test_operations.py]: Nit: to reduce nesting, these context managers can all be invoked in the same `with` statement a la:

    with x() as ctx, \
         y() as ctx2, \
         z() as ctx3(():

etc.

File: dss/operations/ssm_params.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+
+@ssm_params.action(
+    "list",
+    arguments={
+        "--json": dict(default=False, action="store_true", help="format the output as JSON")
+    },
+)
+def ssm_list(argv: typing.List[str], args: argparse.Namespace):
+    """Print out all variables stored in the SSM store"""
```
Comment (xbrianh) [inline dss/operations/ssm_params.py]: Here and throughout this PR, docstrings could be expanded to indicate which SSM store we're dealing with:
```
"""Print out all variables stored in the SSM store $DSS_DEPLOYMENT_STAGE/environment"""
```

File: dss/operations/ssm_params.py [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+    environment[env_var] = value
+    set_ssm_environment(environment)
+    print(f"Success! Set variable in SSM parameter store environment:")
+    print(f"Name: {env_var}")
+    print(f"Value: {value}")
+    if prev_value:
+        print(f"Previous value: {prev_value}")
+
+
+def unset_ssm_parameter(env_var: str) -> None:
```
Comment (xbrianh) [inline dss/operations/ssm_params.py]: nit: This and `set_ssm_parameter` are only used in one place. Consider inlining them into the respective operations methods.

File: dss/operations/ssm_params.py (hunk 2 above)
Comment (chmreid) [reply dss/operations/ssm_params.py]: No, just some inefficient code. Thank you for the suggestion!

File: tests/__init__.py (hunk 3 above)
Comment (chmreid) [reply tests/__init__.py]: Forgot to do that, fixed!

File: tests/test_operations.py (hunk 5 above)
Comment (chmreid) [reply tests/test_operations.py]: Very nice!

File: tests/__init__.py (hunk 4 above)
Comment (chmreid) [reply tests/__init__.py]: Great! Got rid of `self._stdin` and using `sys.__stdin__` instead.

File: dss/operations/ssm_params.py (hunk 6 above)
Comment (chmreid) [reply dss/operations/ssm_params.py]: Updated - thanks for the suggestion! I wasn't too happy with "SSM store" or "SSM param store environment" so this works better.

File: dss/operations/ssm_params.py (hunk 7 above)
Comment (chmreid) [reply dss/operations/ssm_params.py]: Done! thanks for pointing this out

**Pull request merged**